backend/inspirehep/disambiguation/tasks.py

```diff
@@ -114,19 +114,32 @@ def assign_reference_to_author_if_unambiguous_literature_author_match(
     )
     if matched_ref:
         author_ids = get_value(
-            matched_record_list[0], "inner_hits.authors.hits.hits._source.ids", []
+            matched_record_list[0], "inner_hits.authors.hits.hits[0]._source.ids", []
         )
         assign_bai_to_author(author_ids, author)
         return matched_ref
 
 
-def assign_bai_to_author(author_ids, author):
-    author_bai_list = get_values_for_schema(author_ids, "INSPIRE BAI")
-    if author_bai_list and "INSPIRE BAI" not in get_value(author, "ids.schema", []):
-        author["ids"] = [
-            *author.get("ids", []),
-            {"schema": "INSPIRE BAI", "value": author_bai_list[0]},
-        ]
+def assign_bai_to_author(matched_author_ids, literature_author):
+    matched_author_bai_list = get_values_for_schema(matched_author_ids, "INSPIRE BAI")
+    if not matched_author_bai_list:
+        return
+    literature_author["ids"] = [
+        *literature_author.get("ids", []),
+        {"schema": "INSPIRE BAI", "value": matched_author_bai_list[0]},
+    ]
+
+
+def remove_author_bai_from_id_list(literature_author):
+    literature_author_ids = get_value(literature_author, "ids", [])
+    literature_author_bai_dict = [
+        bai_dict
+        for bai_dict in literature_author_ids
+        if bai_dict["schema"] == "INSPIRE BAI"
+    ]
+    if literature_author_bai_dict:
+        literature_author_ids.remove(literature_author_bai_dict[0])
```

Comment:
don't do this. You're mutating the record, which might have unintended consequences.
Unless I'm misunderstanding something, I believe this whole function can be replaced by
```python
def author_ids_without_bais(literature_author):
    literature_author_ids = get_value(literature_author, "ids", [])
    return [
        bai_dict
        for bai_dict in literature_author_ids
        if bai_dict["schema"] != "INSPIRE BAI"
    ]
```